kpi/serializers.py

```diff
@@ -529,6 +530,28 @@ class AssetSerializer(serializers.HyperlinkedModelSerializer):
     # Only add link instead of hooks list to avoid multiple access to DB.
     hooks_link = serializers.SerializerMethodField()
 
+    def get_permissions(self, obj):
+        queries = []
+        for user_id, perm_id in list(obj.get_effective_perms()):
+            queries.append(Q(permission_id=perm_id, user_id=user_id))
```

Comment:
I don't think this is going to be workable in a production environment, sadly :(
Using a test account in my local environment, Django Debug Toolbar shows the number of queries jumping from 31 to 553; the SQL execution time increases from 41 to 366 ms (~9x), and the overall CPU time goes up about 4.7x.

I'm going to look at the permission code overall and see if we can't avoid calling `get_effective_perms()` on every API request.

![image](https://user-images.githubusercontent.com/2085013/49835925-cea5c380-fd6e-11e8-8ce6-9a94618fd5e3.png)

PS: Getting this info while running inside a Docker container requires setting `INTERNAL_IPS` in `settings.py`; in my case, I used `INTERNAL_IPS=['172.17.0.1']`, which corresponds to the IP assigned to the `docker0` interface on my host:
```
linux-host$ ip addr show docker0
6: docker0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default 
    link/ether 02:42:12:30:0f:ac brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 brd 172.17.255.255 scope global docker0
       valid_lft forever preferred_lft forever
    inet6 fe80::42:12ff:fe30:fac/64 scope link 
       valid_lft forever preferred_lft forever
```

Reply:
Update: the new caching improved the situation, but I'm surprised that it's still twice as slow as before. Using `master`, on a local account with 56 assets:
![before](https://user-images.githubusercontent.com/2085013/57706378-9c590480-7633-11e9-86cb-5b6368585d60.png)

Using this branch, with the new caching commits:
![after](https://user-images.githubusercontent.com/2085013/57706440-bf83b400-7633-11e9-93e6-c5c237fd0077.png)

Reply:
Yep. I know. 
AFAIR  there are lots of repeating queries (because of permissions) that are deeply hidden in Django framework. (retrieving same users several time).
We might be able to optimize it a little bit more but this will be deprecated with `v2`.